File: 1-two-sum/two-sum.js
// Hash Map
// Time and Space: O(n)
var twoSum = function(nums, target) {
    const map = new Map();
    for (let i = 0; i < nums.length; i++) {
        const n = nums[i];
        if (map.has(target - n)) return [i, map.get(target-n)]
        map.set(n, i)
    }
};

File: 1-two-sum/two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var numMap:[Int:Int] = [:]

        for (i, n) in nums.enumerated() {
            if var idx = numMap[target - n] {
                return [idx, i]
            }
            numMap[n] = i
        }
        return []
    }
}